sentinelhub/sentinelhub_request.py

```diff
@@ -10,6 +10,18 @@
 from .time_utils import parse_time_interval
 
 
+def _update_other_args(dict1, dict2):
+    """
+    Function for a recursive update of `dict1` with `dict2`. The function loops over the keys in `dict2` and
+    only the non-dict like values are assigned to the specified keys.
+    """
+    for key, value in dict2.items():
+        if isinstance(value, dict):
+            _update_other_args(dict1[key], value)
+        else:
+            dict1[key] = value
```

Comment:
There are 2 cases where this would fail:

- `value` is a dictionary but `key` doesn't exist in `dict1`
- `value` is a dictionary but `dict1[key]` is not a dictionary

I think the first one could at some point actually happen. Therefore I propose changing L19 into

```Python
if isinstance(value, dict) and isinstance(dict1.get(key), dict):
```

Reply:
Probably the second case is irrelevant in this case, or at least I don't see any cases where this could happen in the scope of general usage.

Regarding the first case you are right, but I changed it in the following way, so that if a key is not present in `dict1`, it is added to `dict1` regardless of the fact if he value is a dictionary or not

```
    for key, value in dict2.items():
        if key in dict1.keys():
            if isinstance(value, dict):
                _update_other_args(dict1[key], value)
            else:
                dict1[key] = value
        else:
            dict1[key] = value
```

Reply:
This fixed the failing test, but there is one other test which fails already on develop, perhaps something changed in the mean time. I'll update the values.

Reply:
Yes, please update values for that one.

Regarding the change in `_update_other_args`, if you join both `if` sentences in your proposed solution you get:

```Python
    for key, value in dict2.items():
         if isinstance(value, dict) and key in dict1:
             _update_other_args(dict1[key], value)
        else:
            dict1[key] = value
```

Then the only difference is in `isinstance(dict1.get(key), dict)` vs `key in dict1`. The first option also covers a case when `dict1[key]` exists but is not a dictionary.

Reply:
Ah.. sorry, I'm a bit slow today. Thanks!

Reply:
Fixed. I also added a new test which normally fails at such resolution if the preview mode is not correct.